src/openrct2/interface/Viewport.cpp

```diff
@@ -1619,21 +1619,20 @@ static void set_interaction_info_from_paint_session(paint_session* session)
  * viewport: edi
  */
 void get_map_coordinates_from_pos(
-    int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y, int32_t* interactionType,
-    TileElement** tileElement, rct_viewport** viewport)
+    ScreenCoordsXY screenCoords, int32_t flags, CoordsXY& mapCoords, int32_t* interactionType, TileElement** tileElement,
+    rct_viewport** viewport)
 {
-    rct_window* window = window_find_from_point(ScreenCoordsXY(screenX, screenY));
-    get_map_coordinates_from_pos_window(window, screenX, screenY, flags, x, y, interactionType, tileElement, viewport);
+    rct_window* window = window_find_from_point(screenCoords);
+    get_map_coordinates_from_pos_window(window, screenCoords, flags, mapCoords, interactionType, tileElement, viewport);
 }
 
-void get_map_coordinates_from_pos_window(
-    rct_window* window, int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y, int32_t* interactionType,
+void get_map_coordinates_from_pos(
+    int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y, int32_t* interactionType,
     TileElement** tileElement, rct_viewport** viewport)
 {
-    ScreenCoordsXY screenCoords(screenX, screenY);
+    ScreenCoordsXY screenCoords{ screenX, screenY };
```

Comment:
Why not change the function to take ScreenCoordsXY in the first place?

Reply:
Thats what I have done. This is making a shim so that I don't need to change every use of the function in one go. After this is merged I'll start moving the calling functions to use the new form. This function is used for a lot of things ~100+ and it would be very messy to change them all in one go.

Reply:
Doh, I should have paid more attention!

Reply:
The diff made this look odd so not surprised